src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -150,11 +194,6 @@ public static Stream PackSourcesIntoStream(Uri entrypointFileUri, params ISource
 
         private SourceArchive(Stream stream)
```

Comment:
same here on constructing the object with a BinaryData as arg

Reply:
I'm passing in a file stream, and the gzip APIs use streams.